### PR TITLE
Add `MergeExclusive` type

### PR DESCRIPTION
**Pull request by BendingBender**

### CODE REVIEW

**Comment by sindresorhus** [comment]
Can you use a more readable name than `XOR`? (Not everyone knows what that means)

**Comment by sindresorhus** [comment]
This type looks very useful 👍

File: test/xor.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+const exclusiveVariation1: Options = {exclusive1: true};
+const exclusiveVariation2: Options = {exclusive2: 1};
+
+expectType<{option?: string; exclusive1: boolean; exclusive2?: string}>(
+	exclusiveVariation1
+);
+expectType<{option?: string; exclusive1?: string; exclusive2: number}>(
+	exclusiveVariation2
+);
```
Comment (sindresorhus) [inline test/xor.ts]: Can you add some commented-out negative tests.

So we can easily use those when https://github.com/SamVerschueren/tsd-check/issues/2 is fixed.

Example: https://github.com/sindresorhus/type-fest/pull/1/files#diff-fd2873d8135a3976c91e67f021a5cd40R21

**Comment by janriemer** [comment]
> Can you use a more readable name than XOR? (Not everyone knows what that means)

Maybe `MergeExclusive` or `CombineExclusive` is better!? I would prefer `MergeExclusive`, because we already have a `Merge` type.

**Comment by EdwardDrapkin** [comment]
if you implement `Diff` instead of `Without` (return T[key] instead of never in the map), you should be able to make this type `Diff<A, B> & Diff<B,A>` which is much simpler and Diff's a more useful helper than Without is.

**Comment by sindresorhus** [comment]
I just added some contribution guidelines. Would you be able to review them? I'm also interested in feedback if anything there is unclear or could be improved.

https://github.com/sindresorhus/type-fest/blob/master/.github/contributing.md

**Comment by BendingBender** [comment]
@EdwardDrapkin Do you mean like this:
```ts
type Diff<FirstType, SecondType> = {
	[KeyType in Exclude<keyof FirstType, keyof SecondType>]: FirstType[KeyType]
};
type MergeExclusive<FirstType, SecondType> = Diff<FirstType, SecondType> & Diff<SecondType, FirstType>;
```

This, sadly doesn't work, it basically produces `FirstType & SecondType`, there is nothing exclusive about this type.

**Comment by sindresorhus** [comment]
@BendingBender Can you fix the merge conflict?

File: test/merge-exclusive.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+expectType<{option?: string; exclusive1: boolean; exclusive2?: string}>(
+	exclusiveVariation1
+);
+expectType<{option?: string; exclusive1?: string; exclusive2: number}>(
+	exclusiveVariation2
+);
+
+// TODO: Negative tests, waiting for https://github.com/SamVerschueren/tsd-check/issues/2
+// expectError<Options>({exclusive1: true, exclusive2: 1});
```
Comment (SamVerschueren) [inline test/merge-exclusive.ts]: Upgrade to `tsd-check@5.0.0` and this should work 👌 

```ts
expectError<Options>({exclusive1: true, exclusive2: 1});
```

**Comment by BendingBender** [comment]
Waiting for https://github.com/SamVerschueren/tsd-check/pull/16

**Comment by BendingBender** [comment]
@sindresorhus I think that something is off with the way travis integration is configured. Currently, two different travis runs are spawned up on every commit.

**Comment by sindresorhus** [comment]
> Currently, two different travis runs are spawned up on every commit.

For some reason, there was a webhook to the old `.org` version of Travis too.

**Comment by sindresorhus** [comment]
Thanks :)

**Pull request merged**